src/olympia/abuse/forms.py

```diff
@@ -1,23 +1,79 @@
 from django import forms
 from django.utils.translation import gettext_lazy as _
 
+from olympia.api.throttling import (
+    CheckThrottlesFormMixin,
+    GranularIPRateThrottle,
+    GranularUserRateThrottle,
+)
 
-class AbuseAppealEmailForm(forms.Form):
+
+class HourlyAbuseAppealUserThrottle(GranularUserRateThrottle):
+    rate = '6/hour'
+    scope = 'hourly_user_abuse_appeal'
+
+
+class HourlyAbuseAppealIPThrottle(GranularIPRateThrottle):
+    rate = '6/hour'
+    scope = 'hourly_ip_abuse_appeal'
+
+
+class AbuseAppealUserThrottle(GranularUserRateThrottle):
+    rate = '20/day'
+    scope = 'daily_user_abuse_appeal'
+
+
+class AbuseAppealIPThrottle(GranularIPRateThrottle):
+    rate = '20/day'
+    scope = 'daily_ip_abuse_appeal'
+
+
+class AbuseAppealEmailForm(CheckThrottlesFormMixin, forms.Form):
     # Note: the label is generic on purpose. It could be an appeal from the
     # reporter, or from the target of a ban (who can no longer log in).
     email = forms.EmailField(label=_('Email address'))
 
+    throttled_error_message = _(
+        'You have submitted this form too many times recently. '
+        'Please try again after some time.'
+    )
+
+    throttle_classes = (
+        HourlyAbuseAppealIPThrottle,
+        AbuseAppealIPThrottle,
+    )
+
     def __init__(self, *args, **kwargs):
-        self.expected_email = kwargs.pop('expected_email')
+        self.request = kwargs.pop('request')
+        self.expected_email = kwargs.pop('expected_email', None)
         return super().__init__(*args, **kwargs)
 
     def clean_email(self):
-        if (email := self.cleaned_data['email']) != self.expected_email:
+        if (
+            self.expected_email is None
```

Comment:
Given this can only happen when the kwarg to `__init__` is missing or `None` shouldn't we be erroring further up or raising as a server error?

Reply:
Yeah, let's do that 75d0dfb7039b711a8b059430f63e65d438448adb